### PR TITLE
Atmosphere cas added

### PR DESCRIPTION
**Pull request by areysset**
The method calculating CAS in atmosphere class has been added.
The corresponding test has also been added.

### CODE REVIEW

File: src/fastoad/model_base/atmosphere.py [hunk 1]
```diff
@@ -182,6 +183,23 @@ def true_airspeed(self) -> Union[float, Sequence[float]]:
...
+                        / 5.0
+                        + 1.0
+                    )
+                    ** 3.5
+                    - 1.0
+                )
+                total_pressure = current_level.pressure + impact_pressure
+                sigma_0 = total_pressure / current_level.pressure
+                gamma = 1.4
+                mach = (2.0 / (gamma - 1.0) * (sigma_0 ** ((gamma - 1.0) / gamma) - 1.0)) ** 0.5
```
Comment (christophe-david) [inline src/fastoad/model_base/atmosphere.py:201]: Here your formula explicitly uses gamma, but in the formula above, you use `3.5`, which is &#947;/(&#947;-1). For sake of consistency, the `gamma` variable should be used everywhere, or nowhere. And for sake of efficiency, I think that nowhere is the best choice.

Therefore, I suggest:
```suggestion
                mach = (5.0 * (sigma_0 ** (1.0 / 3.5) - 1.0)) ** 0.5
```

**Review by christophe-david — CHANGES_REQUESTED** [review]
Thank you for this proposal.

As you will see, I have some comments, but the most important one is that you implemented a conversion that is valid only for subsonic flows.

Indeed, our current applications are at most transonic.
Indeed, the TAS<->CAS conversion in supersonic flows is [trickier](https://en.wikipedia.org/wiki/Calibrated_airspeed) 
And indeed, handling these two cases while taking into account the possible array-shape of inputs needs attention.

But the Atmosphere class should aim at being a solid and "universal" basis, and personally, knowing that EAS is close enough to CAS for the subsonic domain (considering the needed accuracy for conceptual studies), I prefer having no CAS computation than having a half-baked one.

Yet, I do not doubt that CAS computation is needed to have a complete Atmosphere class, but this computation should be reliable even above Mach 1.0.

**Comment by areysset** [comment]
The formula has been changed to take into account [supersonic flight](https://en.wikipedia.org/wiki/Equivalent_airspeed)!

**Comment by christophe-david** [comment]
> The formula has been changed to take into account [supersonic flight](https://en.wikipedia.org/wiki/Equivalent_airspeed)!

Ok, so it looks like my suggestion was a bad one, since results get further from reference values.
We will have to use the hard way, then...

**Comment by areysset** [comment]
The new formulas taken from [Ref](https://rosap.ntl.bts.gov/view/dot/8909) highligh lower relative errors to reference results.
Also, we are not sure at which point reference results from [URL](http://www.aerospaceweb.org/design/scripts/atmosphere/) are 100% correct (error = 0%)?

**Comment by christophe-david** [comment]
Another point: given your last modifications, I think you may have not read all my initial comments about gamma, and the mach-related code lines, did you ?
I can also add another point I should have spotted earlier: you can replace all your occurences of `current_level` by `self`.

**Comment by areysset** [comment]
> Another point: given your last modifications, I think you may have not read all my initial comments about gamma, and the mach-related code lines, did you ?
> I can also add another point I should have spotted earlier: you can replace all your occurences of `current_level` by `self`.

I did miss the comment about gamma, sorry for that. I'll remove it definitly.
I'm gonna replace current_level then...

**Comment by christophe-david** [comment]
Version 0.2.0 of [StdAtm](https://pypi.org/project/stdatm/) proposes a computation of calibrated airspeed, valid at subsonic and supersonic speeds, (and also dynamic and impact pressures).

**Pull request closed without merging**